pkg/preparer/orchestrate_test.go

```diff
@@ -89,7 +95,7 @@ func (f *fakeHooks) RunHookType(hookType hooks.HookType, pod hooks.Pod, manifest
 		f.ranAfterLaunch = true
 		return f.afterLaunchErr
 	case hooks.AFTER_AUTH_FAIL:
-		f.ranAfterAuthFail = true
+		f.ranAfterAuthFail <- struct{}{}
```

Comment:
I'm worried that this will be a gotcha for people writing new tests. If their test case calls the after auth fail hook then they can deadlock their test